File: cpp-pr/46/cpp/code-quality-error-handling/file_processor.cpp
#include <iostream>
#include <fstream>
#include <string>

class FileProcessor {
public:
    std::string readFile(const std::string& filename) {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        std::ifstream file(filename);
        std::string content;
        std::string line;
        while (std::getline(file, line)) {
            content += line + "\n";
        }
        return content;
        // {/fact}
    }
};

int main() {
    FileProcessor processor;
    std::string data = processor.readFile("data.txt");
    std::cout << data << std::endl;
    return 0;
}

File: csharp-pr/47/csharp/code-quality-error-handling/FileProcessor.cs
using System;
using System.IO;

namespace FileOperations
{
    public class FileProcessor
    {
        public string ReadFileContent(string filePath)
        {
            // {fact rule=code-quality-error-handling@v1.0 defects=0}
            try
            {
                return File.ReadAllText(filePath);
            }
            catch (FileNotFoundException ex)
            {
                Console.WriteLine($"File not found: {ex.Message}");
                return string.Empty;
            }
            catch (UnauthorizedAccessException ex)
            {
                Console.WriteLine($"Access denied: {ex.Message}");
                return string.Empty;
            }
            catch (IOException ex)
            {
                Console.WriteLine($"IO error: {ex.Message}");
                return string.Empty;
            }
            // {/fact}
        }
    }
}